### ISSUE TITLE
Color Picker Widget

### ISSUE THREAD

**Issue by VisenDev** [david-vanderson/dvui#78]
Add support for a specialized color picker widget similar to the following.

<img width="364" alt="Screenshot 2024-07-29 at 11 58 15 AM" src="https://github.com/user-attachments/assets/ed63701c-778e-4a70-81c1-6715f0478e8d">

.
The current implementation of a basic color picker in `src/Example.zig` looks like this

```zig
// Let's wrap the sliderEntry widget so we have 3 that represent a Color
pub fn rgbSliders(src: std.builtin.SourceLocation, color: *dvui.Color, opts: Options) !void {
    var hbox = try dvui.box(src, .horizontal, opts);
    defer hbox.deinit();

    var red: f32 = @floatFromInt(color.r);
    var green: f32 = @floatFromInt(color.g);
    var blue: f32 = @floatFromInt(color.b);

    _ = try dvui.sliderEntry(@src(), "R: {d:0.0}", .{ .value = &red, .min = 0, .max = 255, .interval = 1 }, .{ .gravity_y = 0.5 });
    _ = try dvui.sliderEntry(@src(), "G: {d:0.0}", .{ .value = &green, .min = 0, .max = 255, .interval = 1 }, .{ .gravity_y = 0.5 });
    _ = try dvui.sliderEntry(@src(), "B: {d:0.0}", .{ .value = &blue, .min = 0, .max = 255, .interval = 1 }, .{ .gravity_y = 0.5 });

    color.r = @intFromFloat(red);
    color.g = @intFromFloat(green);
    color.b = @intFromFloat(blue);
}
```

**Comment by david-vanderson** [comment]
Oh that would be cool.  Can you point me to the place (raylib I assume) that implements that color picker?

**Comment by VisenDev** [comment]
> Can you point me to the place (raylib I assume) that implements that color picker?

The color picker functions are implemented around line 3500 in [raygui.h](https://github.com/raysan5/raygui/blob/master/src/raygui.h)

**Comment by VisenDev** [comment]
After a little experimentation, I have discovered that dvui does have the capability to draw gradient rectangles. 

![Screenshot_20240803_162749](https://github.com/user-attachments/assets/6bc3cd0f-871c-4ba9-803d-52b8191ea3ee)

This is an extremely simple way to do that, but I wanted to verify that it was even possible because I was not really familiar with how `renderGeometry` worked

**Comment by VisenDev** [comment]
As for the hue selector bar, theoretically the simplest way to do that is just to embed a pregenerated image and use it as the background for a slider.

**Comment by david-vanderson** [comment]
Excellent!  That code looks great, let me play around with it a little bit to make sure the function signature feels right to call.

I looked at how raygui does the hue bar - they split the bar into 6 parts and draw a gradient rectangle for each.  So that seems like a good strategy.

**Comment by VisenDev** [comment]
> Excellent! That code looks great, let me play around with it a little bit to make sure the function signature feels right to call.

Yes, the implementation details and signatures should probably change. I was just interested if `renderGeometry` even supported this before I started experimenting with the `pathFillConvex` functions and the like

**Comment by VisenDev** [comment]
Very nice! Thanks for implementing this
